apps/webapp/src/utils/commit-hash-vite-plugin.ts

```diff
@@ -0,0 +1,44 @@
+import { Plugin } from 'vite';
+import { execSync } from 'child_process';
+import { appendFileSync, existsSync, readFileSync } from 'fs';
+
+const PATH_TO_DECLARATION_FILE = 'src/vite-env.d.ts';
+const COMMIT_VAR_DECLARATION = `declare const __COMMIT_HASH__: string;\n`;
+const COMMIT_DATE_DECLARATION = `declare const __COMMIT_DATE__: string;\n`;
+
+// Vite plugin used to inject the current commit hash + commit date into React
+// so the user can be informed the version of minifront they are using
+export const commitHashPlugin = (): Plugin => {
+  const commitHash = execSync('git rev-parse HEAD').toString().trim();
+  const commitDate = execSync('git log -1 --format=%cI').toString().trim();
```

Comment:
Will this work for a deploy to firebase?

Reply:
Indeed, here's an example of the build output with the commit in the bundle (tested in github docker container)

<img width="746" alt="Screenshot 2024-03-01 at 11 44 54 AM" src="https://github.com/penumbra-zone/web/assets/16624263/c807b273-0241-4818-919f-536d7eca7996">